feat: Adding a new param for explicitly setting slot names.

ocworld opened the pull request:
* email: keunhyun.oh@ahnlab.com

Feat: Adding a new param for explicitly setting slot names.

It is related to #740 

Though it is possible to assign feature names to dataset by using spark dataframe's meta in mmlspark, sometimes, I want to set names more simply without handling spark dataframe because of several reasons.

If a developer set the slotNames param, feature names are set to them. If not, it is referenced to spark dataframe's meta.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Azure/mmlspark?pullRequest=752) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Azure/mmlspark?pullRequest=752) before we can accept your contribution.<br/><br/>:x: ocworld [sign now](https://cla.opensource.microsoft.com/Azure/mmlspark?pullRequest=752)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Azure/mmlspark?pullRequest=752) it.</sub>

mhamilton723 commented:
/azp run

mhamilton723 commented:
@ocworld This seems to break default usage, please add a default so this logic only happens when the user sets it explicitly. Also, could you add a unit test with ideal usage?

ocworld commented:
@mhamilton723 I added setting default for slot names. If the param is not set, the code is operated as same as legacy based on dataframe's meta.

Also, I added a unittest for testing setting slot names. However, though I've tried to launch unittests in mmlspark in several times, it is always failed before entering the test I added. How to run test in my local machine and how to verify my test now?

imatiach-msft commented:
/azp run

imatiach-msft commented on src/main/scala/com/microsoft/ml/spark/lightgbm/TrainUtils.scala:
```diff
@@ -232,18 +232,23 @@ private object TrainUtils extends Serializable {
-          attr.index.foreach(index => colnames(index) = attr.name.getOrElse(index.toString))
+  def getSlotNames(schema: StructType, featuresColumn: String, numCols: Int,
+                   trainParams: TrainParams): Option[Array[String]] = {
+    if(trainParams.featureNames.nonEmpty) {
```
nit: space after if

ocworld replied:
I added a space after if

imatiach-msft commented:
/azp run

ocworld commented:
@imatiach-msft @mhamilton723 I wanna fix failure on "Azure.mmlspark (UnitTests lightgbm1)". However, I cannot test it on local. Can I see the error log?

imatiach-msft commented:
@ocworld seems it is some issue with the slot names test, I can take a look at it (it's really unfortunate that there is no way for you to see the build, maybe @mhamilton723 may have some suggestions):

java.lang.Exception: Dataset set feature names call failed in LightGBM with error: basic_string::_S_construct null not valid
	at com.microsoft.ml.spark.lightgbm.LightGBMUtils$.validate(LightGBMUtils.scala:31)
	at com.microsoft.ml.spark.lightgbm.LightGBMDataset$$anonfun$setFeatureNames$1.apply(LightGBMDataset.scala:94)
	at com.microsoft.ml.spark.lightgbm.LightGBMDataset$$anonfun$setFeatureNames$1.apply(LightGBMDataset.scala:92)
	at scala.Option.foreach(Option.scala:257)
	at com.microsoft.ml.spark.lightgbm.LightGBMDataset.setFeatureNames(LightGBMDataset.scala:92)
	at com.microsoft.ml.spark.lightgbm.LightGBMUtils$.generateSparseDataset(LightGBMUtils.scala:255)
	at com.microsoft.ml.spark.lightgbm.TrainUtils$.generateDataset(TrainUtils.scala:50)
	at com.microsoft.ml.spark.lightgbm.TrainUtils$.translate(TrainUtils.scala:264)
	at com.microsoft.ml.spark.lightgbm.TrainUtils$.trainLightGBM(TrainUtils.scala:416)
	at com.microsoft.ml.spark.lightgbm.LightGBMBase$$anonfun$6.apply(LightGBMBase.scala:154)
	at com.microsoft.ml.spark.lightgbm.LightGBMBase$$anonfun$6.apply(LightGBMBase.scala:154)
	at org.apache.spark.sql.execution.MapPartitionsExec$$anonfun$5.apply(objects.scala:188)
	at org.apache.spark.sql.execution.MapPartitionsExec$$anonfun$5.apply(objects.scala:185)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsInternal$1$$anonfun$apply$24.apply(RDD.scala:836)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsInternal$1$$anonfun$apply$24.apply(RDD.scala:836)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:121)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:408)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:414)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
19/12/06 22:18:29 ERROR Executor: Exception in task 1.0 in stage 1574.0 (TID 1972)
java.lang.Exception: Dataset set feature names call failed in LightGBM with error: basic_string::_S_construct null not valid
	at com.microsoft.ml.spark.lightgbm.LightGBMUtils$.validate(LightGBMUtils.scala:31)
	at com.microsoft.ml.spark.lightgbm.LightGBMDataset$$anonfun$setFeatureNames$1.apply(LightGBMDataset.scala:94)
	at com.microsoft.ml.spark.lightgbm.LightGBMDataset$$anonfun$setFeatureNames$1.apply(LightGBMDataset.scala:92)
	at scala.Option.foreach(Option.scala:257)
	at com.microsoft.ml.spark.lightgbm.LightGBMDataset.setFeatureNames(LightGBMDataset.scala:92)
	at com.microsoft.ml.spark.lightgbm.LightGBMUtils$.generateSparseDataset(LightGBMUtils.scala:255)
	at com.microsoft.ml.spark.lightgbm.TrainUtils$.generateDataset(TrainUtils.scala:50)
	at com.microsoft.ml.spark.lightgbm.TrainUtils$.translate(TrainUtils.scala:264)
	at com.microsoft.ml.spark.lightgbm.TrainUtils$.trainLightGBM(TrainUtils.scala:416)
	at com.microsoft.ml.spark.lightgbm.LightGBMBase$$anonfun$6.apply(LightGBMBase.scala:154)
	at com.microsoft.ml.spark.lightgbm.LightGBMBase$$anonfun$6.apply(LightGBMBase.scala:154)
	at org.apache.spark.sql.execution.MapPartitionsExec$$anonfun$5.apply(objects.scala:188)
	at org.apache.spark.sql.execution.MapPartitionsExec$$anonfun$5.apply(objects.scala:185)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsInternal$1$$anonfun$apply$24.apply(RDD.scala:836)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsInternal$1$$anonfun$apply$24.apply(RDD.scala:836)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:121)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:408)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:414)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
19/12/06 22:18:29 WARN TaskSetManager: Lost task 0.0 in stage 1574.0 (TID 1971, localhost, executor driver): java.lang.Exception: Dataset set feature names call failed in LightGBM with error: basic_string::_S_construct null not valid
	at com.microsoft.ml.spark.lightgbm.LightGBMUtils$.validate(LightGBMUtils.scala:31)
	at com.microsoft.ml.spark.lightgbm.LightGBMDataset$$anonfun$setFeatureNames$1.apply(LightGBMDataset.scala:94)
	at com.microsoft.ml.spark.lightgbm.LightGBMDataset$$anonfun$setFeatureNames$1.apply(LightGBMDataset.scala:92)
	at scala.Option.foreach(Option.scala:257)
	at com.microsoft.ml.spark.lightgbm.LightGBMDataset.setFeatureNames(LightGBMDataset.scala:92)
	at com.microsoft.ml.spark.lightgbm.LightGBMUtils$.generateSparseDataset(LightGBMUtils.scala:255)
	at com.microsoft.ml.spark.lightgbm.TrainUtils$.generateDataset(TrainUtils.scala:50)
	at com.microsoft.ml.spark.lightgbm.TrainUtils$.translate(TrainUtils.scala:264)
	at com.microsoft.ml.spark.lightgbm.TrainUtils$.trainLightGBM(TrainUtils.scala:416)
	at com.microsoft.ml.spark.lightgbm.LightGBMBase$$anonfun$6.apply(LightGBMBase.scala:154)
	at com.microsoft.ml.spark.lightgbm.LightGBMBase$$anonfun$6.apply(LightGBMBase.scala:154)
	at org.apache.spark.sql.execution.MapPartitionsExec$$anonfun$5.apply(objects.scala:188)
	at org.apache.spark.sql.execution.MapPartitionsExec$$anonfun$5.apply(objects.scala:185)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsInternal$1$$anonfun$apply$24.apply(RDD.scala:836)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsInternal$1$$anonfun$apply$24.apply(RDD.scala:836)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:121)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:408)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:414)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

19/12/06 22:18:29 ERROR TaskSetManager: Task 0 in stage 1574.0 failed 1 times; aborting job
[0m[[0m[0minfo[0m] [0m[0m[31m- Verify LightGBM Classifier with slot names parameter *** FAILED ***[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 1574.0 failed 1 times, most recent failure: Lost task 0.0 in stage 1574.0 (TID 1971, localhost, executor driver): java.lang.Exception: Dataset set feature names call failed in LightGBM with error: basic_string::_S_construct null not valid[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at com.microsoft.ml.spark.lightgbm.LightGBMUtils$.validate(LightGBMUtils.scala:31)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at com.microsoft.ml.spark.lightgbm.LightGBMDataset$$anonfun$setFeatureNames$1.apply(LightGBMDataset.scala:94)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at com.microsoft.ml.spark.lightgbm.LightGBMDataset$$anonfun$setFeatureNames$1.apply(LightGBMDataset.scala:92)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at scala.Option.foreach(Option.scala:257)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at com.microsoft.ml.spark.lightgbm.LightGBMDataset.setFeatureNames(LightGBMDataset.scala:92)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at com.microsoft.ml.spark.lightgbm.LightGBMUtils$.generateSparseDataset(LightGBMUtils.scala:255)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at com.microsoft.ml.spark.lightgbm.TrainUtils$.generateDataset(TrainUtils.scala:50)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at com.microsoft.ml.spark.lightgbm.TrainUtils$.translate(TrainUtils.scala:264)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at com.microsoft.ml.spark.lightgbm.TrainUtils$.trainLightGBM(TrainUtils.scala:416)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at com.microsoft.ml.spark.lightgbm.LightGBMBase$$anonfun$6.apply(LightGBMBase.scala:154)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at com.microsoft.ml.spark.lightgbm.LightGBMBase$$anonfun$6.apply(LightGBMBase.scala:154)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at org.apache.spark.sql.execution.MapPartitionsExec$$anonfun$5.apply(objects.scala:188)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at org.apache.spark.sql.execution.MapPartitionsExec$$anonfun$5.apply(objects.scala:185)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsInternal$1$$anonfun$apply$24.apply(RDD.scala:836)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsInternal$1$$anonfun$apply$24.apply(RDD.scala:836)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at org.apache.spark.scheduler.Task.run(Task.scala:121)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:408)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:414)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m	at java.lang.Thread.run(Thread.java:748)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31mDriver stacktrace:[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at org.apache.spark.scheduler.DAGScheduler.org$apache$spark$scheduler$DAGScheduler$$failJobAndIndependentStages(DAGScheduler.scala:1889)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1877)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1876)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1876)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:926)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:926)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at scala.Option.foreach(Option.scala:257)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:926)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  ...[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  Cause: java.lang.Exception: Dataset set feature names call failed in LightGBM with error: basic_string::_S_construct null not valid[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at com.microsoft.ml.spark.lightgbm.LightGBMUtils$.validate(LightGBMUtils.scala:31)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at com.microsoft.ml.spark.lightgbm.LightGBMDataset$$anonfun$setFeatureNames$1.apply(LightGBMDataset.scala:94)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at com.microsoft.ml.spark.lightgbm.LightGBMDataset$$anonfun$setFeatureNames$1.apply(LightGBMDataset.scala:92)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at scala.Option.foreach(Option.scala:257)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at com.microsoft.ml.spark.lightgbm.LightGBMDataset.setFeatureNames(LightGBMDataset.scala:92)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at com.microsoft.ml.spark.lightgbm.LightGBMUtils$.generateSparseDataset(LightGBMUtils.scala:255)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at com.microsoft.ml.spark.lightgbm.TrainUtils$.generateDataset(TrainUtils.scala:50)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at com.microsoft.ml.spark.lightgbm.TrainUtils$.translate(TrainUtils.scala:264)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at com.microsoft.ml.spark.lightgbm.TrainUtils$.trainLightGBM(TrainUtils.scala:416)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  at com.microsoft.ml.spark.lightgbm.LightGBMBase$$anonfun$6.apply(LightGBMBase.scala:154)[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  ...[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[32m+ Test Verify LightGBM Classifier with slot names parameter took 0.703s [0m[0m

imatiach-msft commented:
hmm looking at the stack trace I'm guessing it's an issue in the sparse case

imatiach-msft commented:
/azp run

ocworld commented:
Updated to resolve merge conflicts and fix unittest errors. However, it is needed to launch unittests because it is hard to launch it in my local.

imatiach-msft commented:
/azp run

imatiach-msft commented:
@ocworld I think anyone can launch unit tests by just writing the azp run line above, adding @mhamilton723 .  Feel free to run the build whenever you update the PR.

ocworld commented:
@ocworld @mhamilton723  Can I see logs about two build steps' failure? I have wonder that these errors are raised because of my modifications.

imatiach-msft commented:
@ocworld it looks like an infrastructure issue, the E2E cluster is having some connection issues:
[0m[[0m[0minfo[0m] [0m[0m[31m entity:{"error_code":"INVALID_PARAMETER_VALUE","message":"Cluster 1210-154012-natal587 is terminated or does not exist"} [0m[0m

adding @mhamilton723 who developed the e2e build system and would know much more about this

imatiach-msft commented:
/azp run

imatiach-msft commented:
/azp run

imatiach-msft commented:
/azp run

imatiach-msft commented:
/azp run

imatiach-msft commented:
/azp run

imatiach-msft commented:
/azp run

imatiach-msft commented:
/azp run

imatiach-msft commented:
/azp run

imatiach-msft commented:
/azp run

imatiach-msft commented:
@ocworld sorry can you rebase to latest master - can't merge due to conflicts

imatiach-msft commented:
/azp run

mhamilton723 commented:
/azp run

mhamilton723 commented:
@ocworld 
Some minor style nits then this can go in 


[error] /home/vsts/work/1/s/src/main/scala/com/microsoft/ml/spark/lightgbm/TrainParams.scala:69: File line length exceeds 120 characters
[error] /home/vsts/work/1/s/src/main/scala/com/microsoft/ml/spark/lightgbm/TrainParams.scala:91: File line length exceeds 120 characters
[error] /home/vsts/work/1/s/src/main/scala/com/microsoft/ml/spark/lightgbm/TrainParams.scala:111: File line length exceeds 120 characters

ocworld commented:
@mhamilton723 I've fixed them already after finding errors. Please, retry run pipelines.

imatiach-msft commented:
/azp run

imatiach-msft commented:
/azp run

ocworld commented:
@ahnlab-oss
